components/eamxx/src/diagnostics/aodvis.cpp

```diff
@@ -0,0 +1,64 @@
+#include "diagnostics/aodvis.hpp"
+
+#include <ekat/kokkos/ekat_kokkos_utils.hpp>
+
+namespace scream {
+
+AODVis::AODVis(const ekat::Comm &comm, const ekat::ParameterList &params)
+    : AtmosphereDiagnostic(comm, params) {
+  // Nothing to do here
+}
+
+void AODVis::set_grids(
+    const std::shared_ptr<const GridsManager> grids_manager) {
+  using namespace ekat::units;
+  using namespace ShortFieldTagsNames;
+
+  auto grid             = grids_manager->get_grid("Physics");
+  const auto &grid_name = grid->name();
+
+  const auto nondim = Units::nondimensional();
+
+  m_ncols = grid->get_num_local_dofs();
+  m_nlevs = grid->get_num_vertical_levels();
+  // TODO: don't hardcode this!
+  m_swbnd = 14;
+
+  // Define layouts we need (both inputs and outputs)
+  FieldLayout scalar3d_swband_layout{{COL, SWBND, LEV},
+                                     {m_ncols, m_swbnd, m_nlevs}};
+  FieldLayout scalar1d_layout{{COL}, {m_ncols}};
+
+  // The fields required for this diagnostic to be computed
+  add_field<Required>("aero_tau_sw", scalar3d_swband_layout, nondim, grid_name);
+
+  // Construct and allocate the aodvis field
+  // We are going to assume we have nondim units here for ease
+  FieldIdentifier fid("AerosolOpticalDepth550nm", scalar1d_layout, nondim,
+                      grid_name);
+  m_diagnostic_output = Field(fid);
+  m_diagnostic_output.allocate_view();
+}
+
+void AODVis::compute_diagnostic_impl() {
+  using KT  = KokkosTypes<DefaultDevice>;
+  using MT  = typename KT::MemberType;
+  using ESU = ekat::ExeSpaceUtils<typename KT::ExeSpace>;
+
+  const auto aod = m_diagnostic_output.get_view<Real *>();
+  // TODO: don't hardcode swbnd 10
```

Comment:
Have you thought about exposing the band number as part of the diag API? In practice, it would be the same, but it may be more self-documenting, and less hidden in the code... E.g., you could add this to the class header:
```cpp
  static constexpr int vis_band_idx = 10;
```
and then use it here:
```c++
  const auto tau_vis = ... subfield(1,vis_band_idx) ...
  ```

Reply:
Then it's wise to put it in a header that all three can access. My preference is for `scream_universal_constants.hpp`, even though the file name suggests it should contain model-indep stuff (like "speed of light" kind of stuff), rather than the index of the speed of light.

Another alternative is to use a function, maybe to be put in `src/util/scream_common_physics_functions.hpp`:
```cpp
constexpr int eamxx_vis_band_idx () {
  // Some comment explaining what this is, and that the idx itself
  // may vary if we increase the number/position of bands
  return 10;
}
```